src/services/bundler.ts

```
@@ -1,61 +1,184 @@
+  }
+
+  return Promise.resolve();
 }
 
 export interface BundlerResponse {
   message: string;
-  status: number;
   file?: Promise<Blob>;
 }
 
-export function sendZip(file: File): Promise<BundlerResponse> {
-  const body = new FormData();
-  body.append("content", file);
-  const url = `${import.meta.env.DEV ? "http://localhost:5000" : ""}/data`;
-  const request = fetch(url, {
-    method: "POST",
-    body,
-  });
-  return new Promise((resolve, reject) => {
-    request
-      .then((response) => {
-        if (response.status !== 200) {
-          response.text().then((res) => {
-            reject({ message: res, status: response.status });
-          });
+interface ConfigFile {
+  metadata: {
+    title: string;
+    author: string;
+    description: string;
+    version: string;
+    icons: Record<string, string>;
+  };
+
+  build: {
+    targets: string[];
+    source: string;
+  };
+}
+
+const extensions = {
+  ctr: "3dsx",
+  hac: "nro",
+  cafe: "wuhb",
+};
+
+export async function sendContent(archive: File): Promise<BundlerResponse> {
+  const iconFiles: Record<string, Promise<Blob>> = {};
+  let config: ConfigFile;
+
+  const zip = await JSZip.loadAsync(archive);
+  const content = await zip.files["lovebrew.toml"].async("string");
+
+  const bundle: JSZip = new JSZip();
+
+  let defaultZip: JSZip | null = null;
+  let conversionZip: JSZip | null = null;
+
+  try {
+    config = toml.parse(content);
+
+    // find defined icons
+    for (const key in config.metadata.icons) {
+      if (config.metadata.icons[key] === "") continue;
+
+      const path = config.metadata.icons[key];
+      if (zip.file(path) === null) continue;
+
+      console.log("found icon", key);
+      iconFiles[key] = zip.files[path].async("blob");
+    }
+
+    if (zip.file(new RegExp(`^${config.build.source}/.+`)).length === 0) {
+      return Promise.reject({
+        message: `Source folder '${config.build.source}' not found.`,
+      });
```

Use `throw Error()` instead of `return Promise.reject()`